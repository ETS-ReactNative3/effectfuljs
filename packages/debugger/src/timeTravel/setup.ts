import config from "../config";
config.timeTravel = true;
config.mutationObserver = true;
