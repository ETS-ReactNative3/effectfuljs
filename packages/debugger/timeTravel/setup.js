"use strict";

var _config = require("../config");

_config.default.timeTravel = true;
_config.default.mutationObserver = true;