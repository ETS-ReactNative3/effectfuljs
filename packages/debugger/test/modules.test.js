require("./setup");
const trace = require("./run");
trace.silent = true;
const D = require("../main");

test("modules tracing", function() {
  global.console = { log: jest.fn() };
  const mod = trace(D.evalThunk(require("./__fixtures__/mod1")));
  console.log("> M:", mod);
  const mod1 = trace(D.evalThunk(require("./__fixtures__/mod1")));
  console.log("> M1:", mod1);
  const mod2 = trace(D.evalThunk(require("./__fixtures__/mod2")));
  console.log("> M2:", mod2);
  trace(mod.run());
  expect(console.log.mock.calls).toMatchSnapshot();
});
