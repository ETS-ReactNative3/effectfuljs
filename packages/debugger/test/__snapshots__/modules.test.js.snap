// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modules tracing 1`] = `
Array [
  Array [
    "f1:1@1",
  ],
  Array [
    "A:",
    A {
      "b": Object {
        "b": B {
          "i": 100,
        },
        "cnt": 1,
      },
      "i": 11,
    },
  ],
  Array [
    "f2:1@2",
  ],
  Array [
    "i:2,j:3",
  ],
  Array [
    "f1:1@3",
  ],
  Array [
    "A:",
    A {
      "b": Object {
        "b": B {
          "i": 100,
        },
        "cnt": 3,
      },
      "i": 33,
    },
  ],
  Array [
    "f2:3@4",
  ],
  Array [
    "i:2,j:6",
  ],
  Array [
    "> M:",
    Object {
      "__esModule": true,
      "run": [Function],
    },
  ],
  Array [
    "> M1:",
    Object {
      "__esModule": true,
      "run": [Function],
    },
  ],
  Array [
    "> M2:",
    Object {
      "__esModule": true,
      "f1": [Function],
      "f2": [Function],
      Symbol(@effectful/serialization/descriptor): Object {
        "create": [Function],
        "default$": false,
        "name": "./mod2#exports",
        "overrideProps": Object {
          Symbol(@effectful/serialization/descriptor): false,
        },
        "props": false,
        "read": [Function],
        "readContent": [Function],
        "value": [Circular],
        "write": [Function],
      },
    },
  ],
  Array [
    "f1:1@5",
  ],
  Array [
    "A:",
    A {
      "b": Object {
        "b": B {
          "i": 100,
        },
        "cnt": 5,
      },
      "i": 55,
    },
  ],
  Array [
    "f2:5@6",
  ],
  Array [
    "i:2,j:9",
  ],
]
`;
